CMakeLists.txt

```diff
@@ -71,8 +66,9 @@ option(ENABLE_VIRTUAL_OPERATIONS "This option usually works with REGENERATE_CLIE
 set(BUILD_ONLY "" CACHE STRING "A semi-colon delimited list of the projects to build")
 set(CPP_STANDARD "11" CACHE STRING "Flag to upgrade the C++ standard used. The default is 11. The minimum is 11.")
 set(BUILD_ONLY "lambda" "sns")
-set(BUILD_DEPS ON)
 set(AUTORUN_UNIT_TESTS OFF)
+set(ENABLE_TESTING OFF)
```

Comment:
i am curious why do we choose to embed aws-sdk-cpp as a submodule instead of adding it as an external project? this decision impacts lots of the changes we have in this changeset.

Reply:
by "external module" you mean to install it from "install-deps.sh"?
if so, then, sadly, aws-sdk-cpp is **not** packaged for: centos, rhel, fedora, ubuntu, opensuse

Reply:
sorry for the confusion, i meant to say "ExternalProject".  see https://cmake.org/cmake/help/latest/module/ExternalProject.html

Reply:
if we go for static linking, and no "make install" is needed, then this probably works.
only issue is this requires URL/GIT download of the code while running cmake.
currently our opensuse CI does not have network access while packaging. see: https://tracker.ceph.com/issues/48817

why is this a better solution then a submodule?

Reply:
i think we've discussed this over IRC. there are two different issues here

- static link versus dynamic link
- ExternalProject versus `add_subdirectory()`

Reply:
@tchaikov,
* static vs. dynamic linking:
in the RGW stand-up meeting we argued that that dynamic linking is preferred over static linking (see @mattbenjamin commenst here: https://github.com/ceph/aws-sdk-cpp/pull/2#discussion_r590359559).
* ExternalProject vs. git submodule:
Using ther "ExtrnalProject" option requires that the build system will have internet access during the build process (this is currently failing in our CI for opensuse builds (see: https://tracker.ceph.com/issues/48817). And also requires adding "git" as a Ceph dependency. Why is an "ExternalProject" better than a git submodule?